screens/VehicleListScreen.tsx

```diff
@@ -7,61 +7,66 @@ import {
   StyleSheet,
 } from "react-native";
 
-import { NavigationContainer } from '@react-navigation/native';
-import { createStackNavigator } from '@react-navigation/stack';
 import { Text, View } from "../components/Themed";
+import fetchAPI from "../fetchApi";
+import { Vehicle } from "../models";
 
-const Item = ({ item, onPress }) => (
+interface ItemProps {
+  item: Vehicle;
+  onPress: () => void;
+}
+
+const Item = ({ item, onPress }: ItemProps) => (
   <TouchableOpacity onPress={onPress} style={[styles.item]}>
     <Text style={[styles.title]}>{item.name}</Text>
   </TouchableOpacity>
 );
 
-export default function TabOneScreen({ navigation }) {
+export default function VehicleListScreen({ navigation }) {
   const [isLoading, setLoading] = useState(true);
   const [data, setData] = useState([]);
   const [count, setCount] = useState(0);
-  const [page, setPage] = useState(1); 
-
-  let nextItemId = 0;
+  const [page, setPage] = useState(1);
 
-  async function fetchAPI(){
-    let url = "https://swapi.dev/api/vehicles/?page=" + String(page);
-    console.log(url);
-    let config = {};
-    const response = await fetch(url);
-    const json = await response.json();
-
-    setData(data.concat(json.results));
-    setCount(json.count);
-    setLoading(false);    
-    setPage(page + 1);   
+  async function loadNewPage() {
+    const results = await fetchAPI.getVehiclesList(page);
+    setData(data.concat(results));
```

Comment:
hooki to podstępne konstrukcje. Funkcja zwracana przez `useState` może przyjmować parametr, który albo jest obiektem albo funkcją. W tym przypadku lepiej zastosować:
```suggestion
    setData(d => d.concat(results));
```

Albo może nawet:
```suggestion
    setData(d => [...d, ...results]);
```